src/test/java/com/google/step/datamanager/VoteManagerDatastoreTest.java

```diff
@@ -36,30 +35,50 @@ public void tearDown() {
   }
 
   @Test
-  public void testVote() {
-    // // original should have 0 votes
+  public void getVotes_returnsZeroAtFirst() {
     assertEquals(manager.getVotes(DEAL_ID_A), 0);
+  }
 
-    // vote count should increase when someone votes
+  @Test
+  public void vote_incrementsVotesCorrectly() {
     manager.vote(USER_ID_A, DEAL_ID_A, 1);
     assertEquals(manager.getVotes(DEAL_ID_A), 1);
     manager.vote(USER_ID_B, DEAL_ID_A, 1);
     assertEquals(manager.getVotes(DEAL_ID_A), 2);
+  }
 
-    // vote count should not increase when same person votes
+  @Test
+  public void vote_doesNotIncrementForDuplicateVoteFromUser() {
     manager.vote(USER_ID_A, DEAL_ID_A, 1);
-    assertEquals(manager.getVotes(DEAL_ID_A), 2);
+    assertEquals(manager.getVotes(DEAL_ID_A), 1);
+    manager.vote(USER_ID_A, DEAL_ID_A, 1);
+    assertEquals(manager.getVotes(DEAL_ID_A), 1);
+  }
 
-    // vote count should decrease when someone votes -1
+  @Test
+  public void vote_decrementVotesCorrectly() {
     manager.vote(USER_ID_A, DEAL_ID_A, -1);
-    assertEquals(manager.getVotes(DEAL_ID_A), 0);
+    assertEquals(manager.getVotes(DEAL_ID_A), -1);
+    manager.vote(USER_ID_B, DEAL_ID_A, -1);
+    assertEquals(manager.getVotes(DEAL_ID_A), -2);
+  }
 
-    // voting on another deal should not affect vote count for first deal
-    manager.vote(USER_ID_C, DEAL_ID_B, -1);
-    assertEquals(manager.getVotes(DEAL_ID_A), 0);
+  @Test
+  public void vote_doesNotAffectOtherDeals() {
+    manager.vote(USER_ID_A, DEAL_ID_A, 1);
+    manager.vote(USER_ID_A, DEAL_ID_B, -1);
+    assertEquals(manager.getVotes(DEAL_ID_A), 1);
     assertEquals(manager.getVotes(DEAL_ID_B), -1);
   }
 
+  @Test
+  public void vote_userChangeDirection() {
```

Comment:
Let's add another test where the vote should be removed? `vote_userUndoesVote()`?

```
manager.vote(USER_ID_A, DEAL_ID_A, 1);
assertEquals(manager.getVotes(DEAL_ID_A), 1);
manager.vote(USER_ID_A, DEAL_ID_A, 0);
assertEquals(manager.getVotes(DEAL_ID_A), 0);
```